clients/venice-producer/src/main/java/com/linkedin/venice/producer/online/ProducerTool.java

```diff
@@ -193,17 +196,23 @@ private static void writeToStore(ProducerContext producerContext) throws Excepti
 
   private static Object getValueObject(String valueString, RouterBasedStoreSchemaFetcher schemaFetcher) {
     Object value = null;
+    List<Exception> exceptionList = new ArrayList<>();
     for (Schema valueSchema: schemaFetcher.getAllValueSchemas()) {
       try {
         value = adaptDataToSchema(valueString, valueSchema);
         break;
       } catch (Exception e) {
-        // Nothing to do. Try the next schema
+        exceptionList.add(e);
+        // Try the next schema
       }
     }
     if (value == null) {
+      String exceptionDelimiter = "\n\t";
+      String exceptionDetails =
+          exceptionList.stream().map(Throwable::getMessage).collect(Collectors.joining(exceptionDelimiter));
```

Comment:
IIRC getMessage could be null for certain types of exceptions eg NPE.

Reply:
Changed the output format to:
```
[ERROR] Value schema not found. Is a schema that is compatible with the specified data already registered?
Exception messages for each schema:
	Schema Id: 1. Exception: org.apache.avro.AvroTypeException: Expected start-union. Got START_ARRAY
	Schema Id: 2. Exception: org.apache.avro.AvroTypeException: Expected start-union. Got START_ARRAY
```